SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1847,29 +1917,33 @@ void USpatialReceiver::ResolveIncomingOperations(UObject* Object, const FUnrealO
 	// TODO: queue up resolved objects since they were resolved during process ops
 	// and then resolve all of them at the end of process ops - UNR:582
 
-	TSet<FChannelObjectPair>* TargetObjectSet = IncomingRefsMap.Find(ObjectRef);
+	TSet<FSpatialObjectRepState*>* TargetObjectSet = ObjectRefToRepStateMap.Find(ObjectRef);
 	if (!TargetObjectSet)
 	{
 		return;
 	}
 
 	UE_LOG(LogSpatialReceiver, Verbose, TEXT("Resolving incoming operations depending on object ref %s, resolved object: %s"), *ObjectRef.ToString(), *Object->GetName());
 
-	for (FChannelObjectPair& ChannelObjectPair : *TargetObjectSet)
+	for (FSpatialObjectRepState* RepState : *TargetObjectSet)
 	{
-		FObjectReferencesMap* UnresolvedRefs = UnresolvedRefsMap.Find(ChannelObjectPair);
-		if (!UnresolvedRefs)
+		if (!RepState->UnresolvedRefs.Contains(ObjectRef))
 		{
 			continue;
 		}
 
-		if (!ChannelObjectPair.Key.IsValid() || !ChannelObjectPair.Value.IsValid())
+		FChannelObjectPair ChannelObjectPair = RepState->GetChannelObjectPair();
+		USpatialActorChannel* DependentChannel = ChannelObjectPair.Key.Get();
+		
+		if (!RepState->GetChannelObjectPair().Value.IsValid())
 		{
-			UnresolvedRefsMap.Remove(ChannelObjectPair);
+			if (DependentChannel->ObjectReferenceMap.Find(ChannelObjectPair.Value))
+			{
+				DependentChannel->ObjectReferenceMap.Remove(ChannelObjectPair.Value);
+			}
```

Comment:
Add cleanup code to SpatialActorChannel::ReplicateActor where it looks for deleted subobjects.